smack-core/src/main/java/org/jivesoftware/smack/packet/AbstractStreamOpen.java

```diff
@@ -0,0 +1,20 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.packet;
+
+public abstract class AbstractStreamOpen implements Nonza {
```

Comment:
This should probably extend/implement `TopLevelStreamElement` and not `Nonza` (since it is not a Nonza).

Also all fields of StreamOpen should be moved into this class, together with the constructors and getters.